### PR TITLE
Added GitHub actions for clippy

### LINKED ISSUE #169: Add clippy to CI
Run `cargo clippy` in CI and fail on warnings.

### PR DESCRIPTION
**Pull request by RishiKumarRay**
Fixed #169

### CODE REVIEW

File: .github/workflows/clippy.yml [hunk 1]
```diff
@@ -0,0 +1,17 @@
+name: Check formatting
```
Comment (gussmith23) [inline .github/workflows/clippy.yml:1]:
```suggestion
name: Run Clippy
```

File: .github/workflows/clippy.yml [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+
+jobs:
+  check-formatting:
```
Comment (gussmith23) [inline .github/workflows/clippy.yml:10]:
```suggestion
  run-clippy:
```

File: .github/workflows/clippy.yml [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: Check formatting
+      run: cargo clippy
```
Comment (gussmith23) [inline .github/workflows/clippy.yml:17]:
```suggestion
    - run: cargo clippy
```

**Review by gussmith23 — CHANGES_REQUESTED** [review]

**Comment by RishiKumarRay** [comment]
@gussmith23 please review now

**Comment by gussmith23** [comment]
This is great. Thank you so much. I'll merge this soon -- I'm working on another PR right now which I'd like to merge first, and I'd like to fix the errors thrown by clippy before merging this anyway.

Thank you again!

**Comment by gussmith23** [comment]
Btw, if you wanted to work on another PR, you could start fixing the issues that clippy is raising!

**Review by gussmith23 — APPROVED** [review]

**Pull request merged**